src/Components/CreateImageWizardV2/steps/Review/ReviewStep.tsx

```diff
@@ -1,24 +1,70 @@
 import React, { useState } from 'react';
 
-import { ExpandableSection, Form, Title } from '@patternfly/react-core';
+import {
+  ExpandableSection,
+  Form,
+  Text,
+  TextContent,
+  TextVariants,
+  Title,
+} from '@patternfly/react-core';
 
 import { ImageOutputList } from './imageOutput';
+import {
+  TargetEnvAWSList,
+  TargetEnvAzureList,
+  TargetEnvGCPList,
+  TargetEnvOciList,
+  TargetEnvOtherList,
+} from './targetEnvironment';
 
 import {
   ArchitectureItem,
   Distributions,
 } from '../../../../store/imageBuilderApi';
+import { EnvironmentStateType } from '../ImageOutput/Environment';
+import { GCPAccountTypes } from '../TargetEnvironment/GCP/GCPTarget';
 
 type ReviewStepPropTypes = {
   release: Distributions;
   arch: ArchitectureItem['arch'];
+  environment: EnvironmentStateType;
+  awsManual: boolean;
+  awsAccountId: string;
+  awsSource: [number, string];
+  gcpAccountType: GCPAccountTypes;
+  gcpAccountEmail: string;
+  gcpDomain: string;
+  azureManual: boolean;
+  azureSource: [number, string];
+  azureTenantId: string;
+  azureSubscriptionId: string;
+  azureResourceGroup: string;
 };
 
-const ReviewStep = ({ release, arch }: ReviewStepPropTypes) => {
+const ReviewStep = ({
+  release,
+  arch,
+  environment,
+  awsManual,
+  awsAccountId,
+  awsSource,
+  gcpAccountType,
+  gcpAccountEmail,
+  gcpDomain,
+  azureManual,
+  azureSource,
+  azureTenantId,
+  azureSubscriptionId,
+  azureResourceGroup,
```

Comment:
Before we continue with the following steps - I won't lie, I'm a bit worried about this. We are on the second step and the lists of props are starting to get longer. Not entirely sure how sustainable those will be in the following steps.

The problem is, I can't really think of any solution I would deem "safer" or "better" right now as I didn't look into state management solutions that are available.

On one hand I feel like most of the code is easy to follow and navigate through (also love the comments a lot!), but on the other it's starting to be a lot to review and wrap you mind around when it comes to all the states that come into play.

I'd like to get another opinion about this as I think it could be both perfectly fine in the long run but it could also easily get out of hand with a ton of props needed for every component to work...

@lucasgarfield @jkozol what do you think?

Reply:
We can use context for that. But it can be postponed up until the end of the refactor.

Reply:
The thing is you have two possibilities for passing states to a children component, either you list them all in function call, or you store them in a context and the children component load them from the context. That's what every state management libraries are doing under the hood.

Having the state management library doesn't remove the need of having to write every call to the states. At some point they have to be explicitly called when you want to access the data. The advantage that you have by listing all of them in the component call is that you can actually document what is needed for every component. Docstring etc.

I agree that the review step in that case is going to have a ridiculous list of parameters. We can make an exception for it with a context. But I don't think it's that bad to have them all listed. Even if that's a hundred parameters.
Should these parameters keep being forwarded deeply down a stack of component, I would go use a context, but our wizard isn't deep.

My main question would be, do you find it more easy to follow the states in the DDF way? Because in the DDF way we're using the form state to access the states.

For me I'm always lost with the DDF fashion. Especially because I don't know if a state is used somewhere else.
Having the explicit declaration makes it easier to track what is used where. You can follow a state to where it comes from, jump to the parent definition, check it's usages and see every component that has access to it. They are all listed in a top down manner.

The way I represent it in my mind is, by explicitely defining the parameters for every component you work as a Tree. With a context library you work more as a graph. And a tree is easier to grasp (IMO).

Reply:
For reference, have a look at https://react.dev/learn/passing-data-deeply-with-context#before-you-use-context

Reply:
> What about a structure with all the values? Could we maybe create an object or something like that and start passing and destructuring that? Or is that a stupid idea?

It's a good idea in some places. Another possibility is to pass jsx objects down to the review step instead of props.

But having complex objects means that we might need to start having reducers to update them

Reply:
I agree that explicitly defining the parameters makes it simpler to follow what is being passed where. But, as the state and steps continue to grow the amount of getters, setters, and params are also going to grow. I think we will quickly hit a point where the passing of props will make our data less clear. 

I think that implementing a Context with Reducers now will save a lot of time later. Even though some of our field's data types are bit complex (filesystem), for the most part each customisation has a value, isValid, and set/get. So, setting up a general reducer pattern should hopefully not get too complicated and would allow us to avoid too much prop passing.

Reply:
The React documentation is great, and in this case I think it actually makes a strong argument for the use of Context+Reducer pattern. From the documentation linked above:

>Managing state: As your app grows, you might end up with a lot of state closer to the top of your app. Many distant components below may want to change it. It is common to [use a reducer together with context](https://react.dev/learn/scaling-up-with-reducer-and-context) to manage complex state and pass it down to distant components without too much hassle.

This is exactly what is happening here. We are ending up with a lot of state at the top of the app, with numerous distant components below it.

I think we should do what the React documentation recommends here, and use a reducer + context pattern to pass the state to children components “without too much hassle”.

Reply:
Well these technical solutions seems to be quit popular for you all, I won't get in the way and have them implemented then. I do think using contexts for these props is a mistake maintenance wise, but only time will tell who's guess is the correct one.

Reply:
Thanks for being willing to change course. This is a really important design decision... both paradigms have advantages and disadvantages, for sure. I think that right now is the appropriate time to experiment. I'm glad we tried the code using prop drilling first... looking forward to seeing how the code will look with useReducer + useContext now.